### PR TITLE
logging(ydb): add data integrity logs to grpc and session actor

### LINKED ISSUE #6048: [[data integrity trails]] log additional tx info in grpc/session actor
- user tx_id
- request type
- ydb tx_id
- grpc response code
- issues
- hash(yql_text)
- ydb tx_id
- tx_control
- has effects

### PR DESCRIPTION
**Pull request by ulya-sidorina**

### CODE REVIEW

File: ydb/core/kqp/session_actor/kqp_session_actor.cpp [hunk 1]
```diff
@@ -394,7 +397,6 @@ class TKqpSessionActor : public TActorBootstrapped<TKqpSessionActor> {
         YQL_ENSURE(QueryState->GetDatabase() == Settings.Database,
                 "Wrong database, expected:" << Settings.Database << ", got: " << QueryState->GetDatabase());
 
-        QueryState->EnsureAction();
```
Comment (SammyVimes) [inline ydb/core/kqp/session_actor/kqp_session_actor.cpp:397]: Is this not needed anymore?

File: ydb/core/kqp/common/kqp_data_integrity_trails.h [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+        LogKeyValue("Type", type, ss, /*last*/ true);
+
+        return ss.Str();
+    };
+
+    LOG_DEBUG_S(ctx, NKikimrServices::DATA_INTEGRITY, log(txType, traceId, txId, shardId));
+}
+
+}
+}
```
Comment (SammyVimes) [inline ydb/core/kqp/common/kqp_data_integrity_trails.h:103]: missing newline

File: ydb/core/kqp/common/kqp_data_integrity_trails.h [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+        LogKeyValue("SessionId", request->Get()->GetSessionId(), ss);
+        LogKeyValue("TraceId", request->Get()->GetTraceId(), ss);
+        LogKeyValue("Action", ToString(request->Get()->GetAction()), ss);
+        LogKeyValue("Type", ToString(request->Get()->GetType()), ss);
+
+        if (request->Get()->HasTxControl()) {
+            LogTxControl(request->Get()->GetTxControl(), ss);
+        }
+
+        std::string queryText;
```
Comment (SammyVimes) [inline ydb/core/kqp/common/kqp_data_integrity_trails.h:46]: Values are filtered out, right? Do we need query text then?

File: ydb/core/grpc_services/grpc_integrity_trails.h [hunk 4]
```diff
@@ -0,0 +1,265 @@
...
+        LogKeyValue("Status", ToString(response->Get()->Status), ss);
+        LogKeyValue("Issues", ToString(response->Get()->Issues), ss, /*last*/ true);
+        return ss.Str();
+    };
+
+    LOG_DEBUG_S(ctx, NKikimrServices::DATA_INTEGRITY, log(traceId, response));
+}
+
+}
+}
```
Comment (SammyVimes) [inline ydb/core/grpc_services/grpc_integrity_trails.h:265]: Missing newline

File: ydb/core/data_integrity_trails/data_integrity_trails.h [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+}
+
+template <class TransactionSettings>
+inline void LogTxSettings(const TransactionSettings& txSettings, TStringStream& ss) {
+    switch (txSettings.tx_mode_case()) {
+        case TransactionSettings::kSerializableReadWrite:
+            LogKeyValue("TxMode", "SerializableReadWrite", ss);
+            break;
+        case TransactionSettings::kOnlineReadOnly:
+            LogKeyValue("TxMode", "OnlineReadOnly", ss);
```
Comment (SammyVimes) [inline ydb/core/data_integrity_trails/data_integrity_trails.h:17]: Hmm, are we logging reads as well? I thought we are only interested in writes.

**Review by SammyVimes — CHANGES_REQUESTED** [review]

**Comment by dcherednik** [comment]
May be add some tests?

File: ydb/core/kqp/common/kqp_data_integrity_trails.h (hunk 3 above)
Comment (ulya-sidorina) [reply ydb/core/kqp/common/kqp_data_integrity_trails.h]: Actually, no. Hashed query text is useless

File: ydb/core/data_integrity_trails/data_integrity_trails.h (hunk 5 above)
Comment (ulya-sidorina) [reply ydb/core/data_integrity_trails/data_integrity_trails.h:17]: Query type is undefined on GRPC level, so we should  log all requests here. But we use different levels of logging for executer/session_actor/grpc:

grpc - DEBUG (all requests)
session_actor - INFO (requests that can potentially contain modifications )
executer - NOTICE (requests with modifications)

**Comment by ulya-sidorina** [comment]
Example: https://gist.github.com/ulya-sidorina/33d0d863cb90710c6299bf5ad112a1c3

File: ydb/core/kqp/ut/data_integrity/ya.make [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+    REQUIREMENTS(ram:12)
+ENDIF()
+
+IF (WITH_VALGRIND)
+    TIMEOUT(3600)
+    SIZE(LARGE)
+    TAG(ya:fat)
+ELSE()
+    TIMEOUT(600)
+    SIZE(MEDIUM)
```
Comment (va-kuznecov) [inline ydb/core/kqp/ut/data_integrity/ya.make:16]: SMALL

**Review by dcherednik — DISMISSED** [review]
LGTM

**Review by va-kuznecov — DISMISSED** [review]

File: ydb/core/kqp/session_actor/kqp_session_actor.cpp (hunk 1 above)
Comment (ulya-sidorina) [reply ydb/core/kqp/session_actor/kqp_session_actor.cpp:397]: This ensure is here https://github.com/ydb-platform/ydb/pull/6049/files#diff-d9749fd0cd55fd4d28688918a2d479a043a4c2d1f56a6124150ff741c0c640a1R77

**Review by va-kuznecov — APPROVED** [review]

**Review by SammyVimes — APPROVED** [review]
LGTM

**Pull request merged**